### ISSUE TITLE
Karnaugh map drives multiplexor(s)

### ISSUE THREAD

**Issue by jgraley** [jgraley/inferno-cpp2v#535]
Following #530 and #534, consider creating multiplexors (or similar) from Karnaugh slices instead of globally across the truth table rows.

Consider TT during solve-for, before it is split into rows for different options. K-slices in that TT can be mapped back to the row/column layout that is used to differentiate between evaluators (rows) and solveables (columns). (In this case a K-slice may become discontinuous unless you use gray codes and total degree <= 4.)

Now, it's something like we need eg "selector" expressions which are like => but allow RHS to be a symbol set/solve etc. If LHS is false, return {}. Then we simplify to:

∪[ ∧(p<sub>eval_fixed</sub>) → ∩(P<sub>solve_fixed</sub>) ]

where
 - Union is over K slices.
 - p<sub>eval_fixed</sub> means _evaluator predicate fixed by the K-slice, or negation_, and
 - P<sub>solve_fixed</sub> means _solution of a solveable predicate fixed by the K-slice, or negation_.
 - That which is not fixed by the K-slice does not appear anywhere in the corresponding term:
   - _that's the point!_

**Comment by jgraley** [comment]
At this point, we may as well Karnugh in the usual way to eg
 
∪[ ∧(p<sub>fixed</sub>) ]

**Comment by jgraley** [comment]
ON HOLD. 

These changes could actually slow down evaluation of solves by eg causing evaluatables and solves to be evaluated more than once. Stop after #534.
